### ISSUE TITLE
Segfault when Prober has no targets

### ISSUE THREAD

**Issue by Joshi425** [prometheus-operator/prometheus-operator#4591]
**What happened?**
I Deployed:
```
apiVersion: monitoring.coreos.com/v1
kind: Probe
metadata:
  name: gitlab
  namespace: prometheus
spec:
  prober:
    url: gitlab.com:443
    path: /-/metrics
    scheme: https
  interval: 5m
  scrapeTimeout: 1m
```
and the prometheus operator crashed.

**Did you expect to see something different?**
yes
**How to reproduce it (as minimally and precisely as possible)**:
apply the Probe
**Environment**

* Prometheus Operator version:

    `quay.io/prometheus-operator/prometheus-operator:v0.54.0`
    
    

* Kubernetes version information:

    `Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:17:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.8", GitCommit:"4a3b558c52eb6995b3c5c1db5e54111bd0645a64", GitTreeState:"clean", BuildDate:"2021-12-15T14:46:22Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}`
    

* Kubernetes cluster kind:
    kubeadm with managed controlplane

* Manifests:

```
apiVersion: monitoring.coreos.com/v1
kind: Probe
metadata:
  name: gitlab
  namespace: prometheus
spec:
  prober:
    url: gitlab.com:443
    path: /-/metrics
    scheme: https
  interval: 5m
  scrapeTimeout: 1m
```

* Prometheus Operator Logs:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1838b57]

goroutine 201 [running]:
github.com/prometheus-operator/prometheus-operator/pkg/prometheus.(*Operator).selectProbes.func2(0xc00415c240, {0x0, 0x0})
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/operator.go:2030 +0xb7
github.com/prometheus-operator/prometheus-operator/pkg/prometheus.(*Operator).selectProbes(0xc0000ffc00, {0x2122010, 0xc00014bb40}, 0xc000f98b00, 0x2956cfe)
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/operator.go:2037 +0xc5c
github.com/prometheus-operator/prometheus-operator/pkg/prometheus.(*Operator).createOrUpdateConfigurationSecret(0xc0000ffc00, {0x2122010, 0xc00014bb40}, 0xc000f98b00, {0xc00502d200, 0x1, 0x1}, 0x0)
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/operator.go:1539 +0x112
github.com/prometheus-operator/prometheus-operator/pkg/prometheus.(*Operator).sync(0xc0000ffc00, {0x2122010, 0xc00014bb40}, {0xc004640150, 0x30})
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/operator.go:1229 +0x63f
github.com/prometheus-operator/prometheus-operator/pkg/prometheus.(*Operator).processNextWorkItem(0xc0000ffc00, {0x2122010, 0xc00014bb40})
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/operator.go:1037 +0x145
github.com/prometheus-operator/prometheus-operator/pkg/prometheus.(*Operator).worker(0xc0003b8840, {0x2122010, 0xc00014bb40})
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/operator.go:1025 +0x39
created by github.com/prometheus-operator/prometheus-operator/pkg/prometheus.(*Operator).Run
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/prometheus/operator.go:434 +0x30f
```

**Comment by simonpasquier** [comment]
Thanks for the report, it is fixed in #4541 and will be available in the next release (v0.55.0) planned for 2022-03-09.

**Comment by PhilipGough** [comment]
Closed by #4541
